actionpack/lib/action_dispatch/railtie.rb

```diff
@@ -39,6 +39,7 @@ class Railtie < Rails::Railtie # :nodoc:
     config.eager_load_namespaces << ActionDispatch
 
     initializer "action_dispatch.configure" do |app|
+      ActionDispatch::Http::URL.default_protocol = "https://" if app.config.force_ssl
```

Comment:
This is a minor nit, and not really a big deal for a web framework technically, but now we're leaking the protocol construction into a Railtie, don't think it should have to know about that at this level of granularity. See my other config suggestion.

Reply:
Ha!  That was the approach I originally took, for the same concerns (https://github.com/rails/rails/commit/a4e4d1b7f30a8a77cea74f28d1308bdf4dadc077).  But then I screamed "YOLO!" and changed it.  I have changed it back as you've suggested.